packages/gatsby/src/utils/node-manifest.ts

```diff
@@ -239,14 +239,28 @@ export async function processNodeManifest(
   const gatsbySiteDirectory = store.getState().program.directory
 
   // write out the manifest file
-  const manifestFilePath = path.join(
+  let manifestFilePath = path.join(
```

Comment:
could we use joinPath here? https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-core-utils/src/path.ts

Reply:
It doesn't look like it handles `:`'s

Reply:
I'm down but I'm curious, what does [this](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-core-utils/src/path.ts#L11) quadruple backslash do for us?

Reply:
Oh yeah, also it won't solve this issue on its own. But if it gives us some extra guardrails for other scenarios using it makes sense

Reply:
Ah, got it, @KyleAMathews I thought you meant can we use it to solve this problem. I see you mean for extra edge-case guards like @veryspry is saying. Makes sense!

Reply:
sorry I should have been more direct — this is the utility we use all over core for creating cross-os paths (https://github.com/gatsbyjs/gatsby/search?q=joinPath) — I forget its whole history but it should be pretty battle-tested now.

I see now better what you're solving e.g. that some manifest IDs can have reserved characters in them. Wouldn't we always need to replace those? There's some others we'd need to handle https://superuser.com/questions/358855/what-characters-are-safe-in-cross-platform-file-names-for-linux-windows-and-os

Reply:
Gotcha! I'll make that change

Yes, we probably do need to handle more characters other than the colon. `:` is probably the most pressing since a node manifest file name will likely be created using a timestamp but I'll look into handling other cases as well,

Reply:
I made the change to use `joinPath` like you suggested.

Also added some more chars to the replace regex to account for other special chars on Windows